src/geogenalg/continuity.py

```diff
@@ -56,3 +57,51 @@ def find_all_endpoints(
             endpoints.append((point, idx, len(entries)))
 
     return endpoints
+
+
+def connect_nearby_endpoints(
+    input_gdf: gpd.GeoDataFrame, gap_threshold: float
+) -> gpd.GeoDataFrame:
+    """Generate helper lines to close small gaps between line endpoints.
+
+    Args:
+    ----
+        input_gdf: A GeoDataFrame containing the original line network
+        gap_threshold: Maximum distance between endpoints to be connected
+
+    Returns:
+    -------
+        A new GeoDataFrame containing only the generated helper lines
+
+    """
+    new_lines = []
+    used_ends = set()
```

Comment:
You could add type hint here just for clarification. Probably (see https://github.com/nlsfi/geogen-algorithms/pull/8/files#r2165658959)
`
used_ends: set[int] = set()
`